### DISCUSSION TITLE
Imported components from Chakra CLI has typescript issues.

### DISCUSSION THREAD

**Discussion by sayshark75** [chakra-ui/chakra-ui#9010] [Q&A] [answered]
- I installed nextjs 14, with chakra ui, here is package.json
```
{
  "name": "test-chakra3",
  "version": "0.1.0",
  "private": true,
  "scripts": {
    "dev": "next dev",
    "build": "next build",
    "start": "next start",
    "lint": "next lint",
    "gen": "chakra typegen src/themes/theme.ts"
  },
  "dependencies": {
    "@chakra-ui/react": "^3.0.0-next.32",
    "@emotion/react": "^11.13.3",
    "next": "14.2.16",
    "next-themes": "^0.3.0",
    "react": "^18",
    "react-dom": "^18",
    "react-icons": "^5.3.0"
  },
  "devDependencies": {
    "@types/node": "^20",
    "@types/react": "^18",
    "@types/react-dom": "^18",
    "eslint": "^8",
    "eslint-config-next": "14.2.16",
    "typescript": "^5"
  }
}
```
when i added accordion component, in that component, there is a ts issue

![image](https://github.com/user-attachments/assets/0d2825d0-f61c-4489-8d24-231d50115b63)

but when i changed it to 
```
type AccordionItemContentProps = Accordion.ItemContentProps;
```
the issue resolved, and i tried deploying it to vercel, works fine, UI rendered and interactive
is there any consequences by using this? or is it fine for now?

- The color mode component also causing same issue, but resolved via this kind of changes.
- if we are just capturing the types, we can directly use it, instead creating a interface.

**Comment by Daidalos117** [comment]
For me it's also happening
<img width="300" alt="Snímek obrazovky 2024-10-28 v 20 53 34" src="https://github.com/user-attachments/assets/3081f512-dcad-4de5-a4e6-727d5b9b9741">

**Comment by segunadebayo** [comment, marked as answer]
Duplicate of https://github.com/chakra-ui/chakra-ui/discussions/9038

Same snippet TypeScript error (`children` / JSX component). Align `@types/react` with your React version, or regenerate snippets on a current CLI.
